OMCompiler/SimulationRuntime/c/CMakeLists.txt

```diff
@@ -38,34 +38,48 @@ MESSAGE(STATUS "${SUNDIALS_INCLUDE_DIR}")
-FIND_LIBRARY(SUNDIALS_LIBRARY_NVEC       NAMES libsundials_nvecserial        PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
-FIND_LIBRARY(SUNDIALS_KINSOL       NAMES libsundials_kinsol        PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_CVODE         NAMES libsundials_cvode                 PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_IDA           NAMES libsundials_ida                   PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_NVEC          NAMES libsundials_nvecserial            PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_KINSOL        NAMES libsundials_kinsol                PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_LINSOL_LAPACK NAMES libsundials_sunlinsollapackdense  PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_LINSOL_KLU    NAMES libsundials_sunlinsolklu          PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_SPGMR         NAMES libsundials_sunlinsolspgmr        PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_SPBCGS        NAMES libsundials_sunlinsolspbcgs       PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
+  FIND_LIBRARY(SUNDIALS_LIBRARY_SPTFQMR       NAMES libsundials_sunlinsolsptfqmr      PATHS /usr/lib /usr/local/lib $ENV{LIB} $(Sundials_Path)/lib)
 endif(MSVC)
 
 IF(SUNDIALS_INCLUDE_DIR)
 
   INCLUDE_DIRECTORIES(${SUNDIALS_INCLUDE_DIR})
 
-  if(SUNDIALS_LIBRARY_CVODE AND
-   SUNDIALS_LIBRARY_IDA AND
-   SUNDIALS_LIBRARY_NVEC AND
-   SUNDIALS_KINSOL)
+  IF(SUNDIALS_LIBRARY_CVODE AND SUNDIALS_LIBRARY_IDA AND SUNDIALS_LIBRARY_NVEC AND
+     SUNDIALS_LIBRARY_KINSOL AND SUNDIALS_LIBRARY_LINSOL_LAPACK AND SUNDIALS_LIBRARY_LINSOL_KLU
+     SUNDIALS_LIBRARY_SPGMR AND SUNDIALS_LIBRARY_SPBCGS AND SUNDIALS_LIBRARY_SPTFQMR)
 
-  SET(SUNDIALS_LIBRARIES ${SUNDIALS_LIBRARY_CVODE} ${SUNDIALS_LIBRARY_IDA} ${SUNDIALS_LIBRARY_NVEC} ${SUNDIALS_KINSOL})
-
-  ENDIF(SUNDIALS_LIBRARY_CVODE AND
-        SUNDIALS_LIBRARY_IDA AND
-        SUNDIALS_LIBRARY_NVEC AND
-        SUNDIALS_KINSOL)
+    SET(SUNDIALS_LIBRARIES ${SUNDIALS_LIBRARY_CVODE} ${SUNDIALS_LIBRARY_IDA} ${SUNDIALS_LIBRARY_NVEC}
+                           ${SUNDIALS_LIBRARY_KINSOL} ${SUNDIALS_LIBRARY_LINSOL_LAPACK} ${SUNDIALS_LIBRARY_LINSOL_KLU}
+                           ${SUNDIALS_LIBRARY_SPGMR} ${SUNDIALS_LIBRARY_SPBCGS} ${SUNDIALS_LIBRARY_SPTFQMR})
+    MESSAGE(STATUS "AHEU: Using following SUNDIALS libraries:")
+    MESSAGE(STATUS "${SUNDIALS_LIBRARIES}")
+    MESSAGE(FATAL_ERROR "AHEU: STOP")
+  ELSE()
+    MESSAGE(FATAL_ERROR "Some SUNDIALS library not found!")
+  ENDIF()
 
 ENDIF(SUNDIALS_INCLUDE_DIR)
+MESSAGE(FATAL_ERROR "AHEU: STOP STOP")
```

Comment:
Some leftover debug code, but is the CMakeFile even used at some point or at least tested?